### PR TITLE
Misc update

### PR DESCRIPTION
**Pull request by iBug**
Round reason weight, strip overzealous type coersion, hide comment
count in reviews

### CODE REVIEW

**Comment by ArtOfCode-** [comment]
status-don't-merge right now. Some comments I want to make when I get some time later

**Comment by angussidney** [comment]
@ArtOfCode- poke

File: app/models/reason.rb [hunk 1]
```diff
@@ -11,13 +11,13 @@ def tp_percentage
 
     count = posts.where(is_tp: true, is_fp: false).count
 
-    (count.to_f / fast_post_count.to_f).to_f
+    count.to_f / fast_post_count.to_f
```
Comment (thesecretmaster) [inline app/models/reason.rb]: Here and elsewhere, you only need `to_f` on one side -- the other side will automatically be coerced to be a float. So, here you could put `count.to_f / fast_post_count` and get the same result.

File: app/views/posts/_post.html.erb [hunk 2]
```diff
@@ -38,7 +38,7 @@
       <span class='text-danger'>(deleted <%= distance_of_time_in_words(post.created_at, post.deleted_at, include_seconds: true) %> after creation)</span>
     <% end %>
 
-    <% if post.comments.any? %>
+    <% if post.comments.any? and not hide_feedbacks %>
```
Comment (thesecretmaster) [inline app/views/posts/_post.html.erb]: Typically in ruby we prefer `&&` for `and` and `!` for `not`, so this really should be `&& !hide_feedbacks` (in terms of code style). Not sure why we want to hide comments if feedbacks are hidden though (or what the `hide_feedbacks` method/variable is).

File: app/helpers/reasons_helper.rb [hunk 3]
```diff
@@ -16,7 +16,7 @@ def self.check_for_inactive_reasons
   def self.calculate_weights_for_flagging
     Reason.all.each do |reason|
       if reason.posts.count > 20
-        reason.update(weight: [reason.tp_percentage * 100, reason.maximum_weight].compact.min)
+        reason.update(weight: [(reason.tp_percentage * 100).round, reason.maximum_weight].compact.min)
```
Comment (thesecretmaster) [inline app/helpers/reasons_helper.rb]: What's the reason for the change here?

**Comment by thesecretmaster** [comment]
@ArtOfCode- Seems good to me, just waiting on your approval to merge.
